docs/develop/python/message-passing.mdx

```diff
@@ -486,6 +484,51 @@ You can silence these warnings on a per-handler basis by passing the `unfinished
 
 See [Finishing handlers before the Workflow completes](/encyclopedia/workflow-message-passing#finishing-message-handlers) for more information.
 
+
+### Use `@workflow.init` to operate on Workflow input before any handler executes
+
+Normally, your Workflow `__init__` method won't have any parameters.
+However, if you use the `@workflow.init` decorator on your `__init__` method, you can give it the same [Workflow parameters](/develop/python/core-application#workflow-parameters) as your `@workflow.run` method.
+The SDK will then ensure that your `__init__` method receives the Workflow input arguments that the  [Client sent](/develop/python/temporal-clients#start-workflow-execution).
+(The Workflow input arguments are also passed to your `@workflow.run` method -- that always happens, whether or not you use the `@workflow.init` decorator.)
+
+To understand why `@workflow.init` is useful, suppose that you've written a Signal or Update handler that does something involving the Workflow input, but that you need to process the workflow input in some way before the handler should be allowed to access it.
+The problem is that it's possible for a Signal or Update handler to start executing _before_ your `@workflow.run` method!
+This means that you can't do the processing in your `@workflow.run` method.
+A solution is to use the `@workflow.init` decorator, and do the processing in your `__init__` method. (An alternative solution would be to [use `@workflow.wait_condition` in your handler](#wait-in-message-handler) to wait until the processing has been done by your `@workflow.run` method.
+That would work, but using `@workflow.init` is simpler.)
```

Comment:
The parentheses here are included after some deliberation, since in general we avoid them. My thinking was that this point is both not-essential to the main argument, and also very desirable to point out. I tentatively concluded that parentheses were a reasonable option here, but feel free to suggest an alternative arrangement.

Reply:
In the overview docs, I disambiguated these two approaches based on whether there's blocking work to be done.  Once again wondering if it's better to just backlink.

Reply:
I agree - we should backlink.

Reply:
Sorry, I should have resolved this thread. It does backlink.